Shake more freely!	Add an option to select shaking mode freely.

wolfcon opened the pull request:
shake more freely!

摇动更加自由顺畅, 向四面八方摇动.

wolfcon committed:
Shake more freely!

poboke commented:
我使用Xcode 7.2 (7C68)，编辑框有很大的错位

![image](https://cloud.githubusercontent.com/assets/4415086/12047219/ee804896-af02-11e5-8800-502902fadb44.png)

poboke commented:
如果是为了实现上下左右震动的话，可以把`[ShakeAction shakeView:textView];`改为`[ShakeAction shakeView:(NSView *)self.containerView];`，这样就不会有错位了。
但是以前我发现这样做的话晃动太大，比较影响体验，所以就去掉了。
不过`arc4random_uniform`这个函数不错，以前没用到过。

wolfcon commented:
我和你使用相同版本的Xcode, 没有出现出现错位的问题啊~

另外改了一下,`[ShakeAction shakeView:(NSView *)self.containerView]`改成这个了

加了一个选项, 上下摇还是四周摇. (总感觉上下摇动不是很舒服, 像被夹住的感觉.)O(∩_∩)O

wolfcon committed:
Add an option to select shaking mode freely.

There are two shaking modes now.
• up and down.
• all direction.

add a gitignore file

wolfcon committed:
Replace particle shape with rectangle in free mode

更换free模式下的颗粒为方块

poboke commented:
很感谢你提交代码，不过还是存在bug的，比如：

```
+        self.soundMenuItem = [self menuItemWithTitle:@"Enable Free Mode  🌀" type:kMenuItemTypeEnableFreeMode];
+        self.soundMenuItem.state = configManager.isEnableSound;
+        self.soundMenuItem.enabled = configManager.isEnablePlugin;
+        [configMenu addItem:self.soundMenuItem];
```

这里的`self.soundMenuItem`忘了改，不过我已经在最新的代码里改正了。

wolfcon commented:
手误, 谢谢了 ,现在感觉性能问题比较严重, 在打字比较快的时候particle的生成会很卡

poboke commented:
是的，你有什么解决方法吗？把震屏关了会好一点。
